cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/write_cloudwatch_agent_json.py

```diff
@@ -177,14 +178,74 @@ def filter_output_fields(configs):
     return [{desired_key: config[desired_key] for desired_key in desired_keys} for config in configs]
 
 
-def create_config(log_configs):
-    """Return a dict representing the structure of the output JSON."""
-    return {
-        "logs": {
-            "logs_collected": {"files": {"collect_list": log_configs}},
-            "log_stream_name": f"{gethostname()}.{{instance_id}}.default-log-stream",
+def select_metrics(args):
+    """Add metrics_collected for CloudWatch Agent Metrics section."""
+    metric_configs = {}
+    metrics_collected = {}
+    metrics_collected = add_metrics_mem(metrics_collected, args.node_role)
+    metrics_collected = add_metrics_disk(metrics_collected, args.node_role)
+    metric_configs["metrics_collected"] = metrics_collected
+    return metric_configs
+
+
+def add_metrics_mem(metrics_collected, node_role):
+    """Add memory metrics for the metrics_collected field of CloudWatch Agent Metrics section."""
+    if node_role == "HeadNode":
```

Comment:
As it is discussed, can we do this with metrics schema like what we do for cloudwatch logs schema so that in the future it will be easier for us to append more metrics?

Reply:
Yes. I moved the information of metrics to the configuration json file, updated the json schema, and renamed the files accordingly (from specific for "log_files" to general for CloudWatch agent).